### ISSUE TITLE
Update mutation breaks composite key

### ISSUE THREAD

**Issue by stanleycyang** [aws-amplify/amplify-cli#1756]
**Describe the bug**
When you run an update mutation on a model with a composite key, it breaks:

<img width="633" alt="Screen Shot 2019-06-30 at 10 42 01 PM" src="https://user-images.githubusercontent.com/3037893/60416143-ef5e2b00-9b91-11e9-8b3a-73ebe146d510.png">

**To Reproduce**

1. Create the following schema

```gql
type Schedule
  @model
  @key(name: "ByProject", fields:  ["projectId", "date", "department"], queryField: "schedulesByProject")
{
  id: ID!
  date: String!
  department: String
  projectId: ID!
  timeCard: TimeCard @connection
  createdAt: AWSDateTime
}

type TimeCard
  @model
{
  id: ID!
  date: AWSDate!
  hours: String
  updatedAt: AWSDateTime
  createdAt: AWSDateTime
}
```

2. create the schedule
3. create the time card, and then run the `updateSchedule` mutation
4. Check the dynamoDB and see the key look like this

<img width="633" alt="Screen Shot 2019-06-30 at 10 42 01 PM" src="https://user-images.githubusercontent.com/3037893/60416334-6dbacd00-9b92-11e9-894d-9ffabe196e7e.png">

**Expected behavior**
The composite key should not be broken when making a simple update

**Comment by mikeparisstuff** [comment]
There is logic that is meant to fail fast when the input object does not contain the @key fields. Can you please include the mutation that is breaking the key to help debug the issue.

**Comment by stanleycyang** [comment]
@mikeparisstuff 

I ran an `updateSchedule` query passing in the `schedule` ID and `scheduleTimeCardId`. I get a success on the mutation, but I'm not quite sure why inserting into the composite sort key the `ctx` would be a fail fast solution.

Here is the actual mutation:

```
mutation UpdateSchedule($input: UpdateScheduleInput!) {
  updateSchedule(input: $input) {
    id
    date
    department
    timeCard {
      id
      date
      hours
    }
  }
}
```

The `UpdateScheduleInput` looks like:

```
input UpdateScheduleInput {
	id: ID!
	date: String
	department: String
	projectId: ID
	createdAt: AWSDateTime
	scheduleTimeCardId: ID
}
```

**Comment by kyangy** [comment]
+1

**Comment by SwaySway** [comment]
Closing this issue as the validation has been added as of version 4.20

**Comment by attilah** [comment]
Validation was only a partial fix, see #3664
